src/main/kotlin/com/bridgecrew/ui/actions/SeverityFilterActions.kt

```diff
@@ -29,7 +32,47 @@ class SeverityFilterActions(val project: Project) : ActionListener {
                 "C" to Severity.CRITICAL
         )
 
+        var enabledSeverities = Severity.values().toList()
+
         var currentSelectedSeverities = Severity.values().toList()
+
+        fun onChangeCategory(category: Category?, project: Project) {
+            updateEnabledSeverities(category, project)
+        }
+
+        fun onSingleFileScanFinished(project: Project) {
+            updateEnabledSeverities(CheckovToolWindowFactory.lastSelectedCategory, project)
+        }
+
+        private fun updateEnabledSeverities(category: Category?, project: Project) {
+            val categoryAsList = if (category == null) null else listOf(category)
+            val categorySeverities = CheckovResultsListUtils.filterResultsByCategoriesAndSeverities(project.service<ResultsCacheService>().checkovResults, categoryAsList, Severity.values().toList()).map{ result -> result.severity}
+            // no pressed category - display the category's severities
+            if (currentSelectedSeverities.size == Severity.values().toList().size) {
+                enabledSeverities = categorySeverities
+                return
+            }
+
+            // there is a pressed category - check if it is included in the category severities - if so - display it, else - severities shouldn't be displayed
+            if (currentSelectedSeverities.any { severity -> categorySeverities.contains(severity) }) {
+                enabledSeverities = categorySeverities
+                return
+            }
+
+            enabledSeverities = listOf()
+        }
+
+        fun restartState() {
```

Comment:
maybe use this function with empty initial state so the initial state is defined only in one place instead of two

Reply:
Because the fields are in companion class they should be inlined initialized anyway 😅